src/components/main/MainCard.tsx

```diff
@@ -26,6 +28,32 @@ const MainCard: FC<MainCardProps> = ({ currentWeather, location }) => {
 
   const date = convertTime(time);
 
+  const getIsFavorite = (): boolean => {
+    const favoriteCities: LocationInfoType[] = getFavoriteCities();
+    return favoriteCities.some(city => city.id === location.id);
+  };
+
+  const [isFavorite, setIsFavorite] = useState(getIsFavorite);
+
+  const toggleFavoriteCities = () => {
+    if (!isFavorite) {
+      const favoriteCities = getFavoriteCities().concat(location);
+      setInLocalStorage(favoriteCities, 'favorite-cities');
+    } else {
+      const favoriteCities = getFavoriteCities().filter(city => city.id !== location.id);
+      setInLocalStorage(favoriteCities, 'favorite-cities');
+    }
```

Comment:
please refactor function:
1. no matter if city is favorite or not, you will use `getFavoriteCities` function to get it, so it can be called once
2. no matter if city is favorite or not, you will use `setInLocalStorage` to set new list, so it also can be called once